File: tests/07_phases_ompss.dg/fortran/success_task_27.f90
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! </testinfo>

PROGRAM P
    IMPLICIT NONE
    INTEGER :: I, LIMIT
        !$OMP TASK FINAL(.FALSE.)
        !$OMP END TASK

        !$OMP TASK FINAL(I < LIMIT)
        !$OMP END TASK

        !$OMP TASKWAIT
END PROGRAM P

File: tests/07_phases_ompss.dg/cxx/success_task_27.cpp
/*
<testinfo>
test_generator=config/mercurium-ompss
</testinfo>
*/
#include<assert.h>

int f(int &x) {
    x = 2;
    return 3;
}

int main(int argc, char*argv[])
{
    int res = 1;
    #pragma omp task shared(res)
    {
        int x = f(x) + x;
        res = x;
    }
    #pragma omp taskwait

    assert(res == 5);
    return 0;
}
